packages/manager/modules/iam/src/routes/onboarding.route.js

```diff
@@ -1,16 +1,33 @@
 import { onboarding as onboardingComponent } from '@iam/components';
-import { asResolve, noBreadcrumbResolve } from '@iam/resolves';
+import {
+  asResolve,
+  noBreadcrumbResolve,
+  advancedModeResolve,
+} from '@iam/resolves';
 
 const name = 'onboarding';
-const resolves = [noBreadcrumbResolve];
+const resolves = [noBreadcrumbResolve, advancedModeResolve];
 
-const state = () => ({
+const state = ({ ROUTES }) => ({
   url: '/onboarding',
   component: onboardingComponent.name,
   resolve: {
     ...asResolve(onboardingComponent.resolves),
     ...asResolve(resolves),
+    resources: /* @ngInject */ (PolicyService, advancedMode) => {
+      return PolicyService.getPolicies({
+        cursor: '',
+        ...(!advancedMode && { readOnly: false }),
+      }).then(({ data }) => data);
+    },
```

Comment:
It is better to have a reusable resolve that returns a boolean, for instance
`hasPoliciesResolve`

Reply:
It's applied